src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -1,173 +1,248 @@
+                               OrdersKafkaProducer kafkaProducer,
+                               KafkaListenerEndpointRegistry registry) {
         this.serializerFactory = serializerFactory;
+        this.kafkaProducer = kafkaProducer;
+        this.registry = registry;
     }
 
-    @Override
-    public void afterPropertiesSet() throws Exception {
-        LOGGER.debug("Initializing kafka consumer service " + this.toString());
-
-        ProducerConfig config = getProducerConfig();
-        CHKafkaProducer chKafkaProducer = new CHKafkaProducer(config);
-
-        ConsumerConfig consumerConfig = getConsumerConfig();
-
-        consumerConfig.setGroupName(ORDER_RECEIVED_GROUP);
-        consumerConfig.setTopics(Collections.singletonList(ORDER_RECEIVED_TOPIC));
-        chKafkaConsumerGroupMain
-                = new CHKafkaResilientConsumerGroup(consumerConfig, CHConsumerType.MAIN_CONSUMER,
-                        new KafkaConsumerFactory(), chKafkaProducer);
-
-        consumerConfig.setTopics(Collections.singletonList(ORDER_RECEIVED_TOPIC_RETRY));
-        consumerConfig.setGroupName(ORDER_RECEIVED_GROUP_RETRY);
-        chKafkaConsumerGroupRetry
-                = new CHKafkaResilientConsumerGroup(consumerConfig, CHConsumerType.RETRY_CONSUMER,
-                        new KafkaConsumerFactory(), chKafkaProducer);
-    }
-
-    @KafkaListener(topics = ORDER_RECEIVED_TOPIC, groupId = ORDER_RECEIVED_GROUP,
-            autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
-    public void processOrderReceived(String orderReceivedUri)
+    /**
+     * `order-received` topic listener/consumer. Calls a `retryable` method to process received message.
+     * If the `retryable` processor is unsuccessful with a `retryable` error, after maximum numbers of attempts allowed,
+     * the message is published to `order-received-retry` topic for failover processing.
+     * @param message
+     * @throws SerializationException
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    @KafkaListener(id = ORDER_RECEIVED_GROUP, groupId = ORDER_RECEIVED_GROUP,
+                    topics = ORDER_RECEIVED_TOPIC,
+                    autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
+    public void processOrderReceived(org.springframework.messaging.Message<OrderReceived> message)
             throws SerializationException, ExecutionException, InterruptedException {
+        OrderReceived orderReceived = message.getPayload();
+        String orderReceivedUri = orderReceived.getOrderUri();
         try {
-            logMessageReceived(orderReceivedUri, ORDER_RECEIVED_TOPIC);
-        } catch (Exception ex){
-            republishMessageToTopic(orderReceivedUri, ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_TOPIC_RETRY, ex);
+            logMessageReceived(message, ORDER_RECEIVED_TOPIC);
+
+            logMessageProcessed(message, ORDER_RECEIVED_TOPIC);
```

Comment:
Message appears to be logged twice.

Reply:
They are two separate messages. The processing of message (sending email) happens between these two messages.